win32ss/gdi/ntgdi/bitmaps.c

```diff
@@ -721,9 +725,17 @@ BITMAP_CopyBitmap(HBITMAP hBitmap)
         return 0;
     }
 
+    bmRealHeight = psurfSrc->SurfObj.sizlBitmap.cy;
+    curIcon = getCurrentIcon();
```

Comment:
this seems like a huge hack?
Just storing a global pointer somewhere, and hoping it does not go invalid / away?

Reply:
@learn-more 
will my last commit resolve your doubts?

Reply:
not really, no